### PR TITLE
Svg icons

### LINKED ISSUE #8064: Apps: support svg icons
The apps page utils module currently sets image type to image/png for all images loaded from disk.  The watcher script currently checks the 64x64 and 128x128 subfolders of the icons dir.  If the watcher also checked the 'scalable' dir and there was some logic added to utils to either check the path or the extension to set the appropriate mime type, vector icons should work.

### PR DESCRIPTION
**Pull request by jeepingben**
Resolves #8064

### CODE REVIEW

File: pkg/apps/watch-appstream.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def try_size(sz):
         path = os.path.join(dir, "..", "icons", origin, sz, icon)
         return path if os.path.exists(path) else None
 
-    return try_size("64x64") or try_size("128x128")
+    return try_size("scalable") or try_size("64x64") or try_size("128x128")
```
Comment (petervo) [inline pkg/apps/watch-appstream.py]: My feeling is that png support is much more straight forward in most browsers so we should prefer png to svg. But maybe that is wrong.

@mvollmer, do you have an opinion?

**Comment by mvollmer** [comment]
I don't think the AppStream spec allows cached icons in the SVG format, and I would be surprised if the extant AppStream collection data generators would ever produce them.

Local, remote, and stock icons can probably be SVGs.

So, the smarter content-type detection is good stuff, but I am hesitant to extend the cached icon search logic.  In fact, that search logic (trying different sizes) might be wrong.  In the words of @hughsie, "it is complicated".

File: pkg/apps/utils.jsx [hunk 2]
```diff
@@ -43,12 +43,18 @@ export function icon_url(path_or_url) {
...
+    } else {
+         contentType = "image/png"
+    }
     var query = window.btoa(JSON.stringify({
         payload: "fsread1",
         binary: "raw",
         path: path_or_url,
         external: {
-            "content-type": "image/png",
+            "content-type": contentType,
```
Comment (mvollmer) [inline pkg/apps/utils.jsx]: Can we let the browser do the content type detection, actually?

File: pkg/apps/watch-appstream.py (hunk 1 above)
Comment (mvollmer) [reply pkg/apps/watch-appstream.py]: Let's not change the cached icon search logic at all, as there is no way in practice to get the collection data generators to create the `/usr/share/app-info/$origin/icons/scalable` directory.  I think.

**Comment by mvollmer** [comment]
@jeepingben, do you have a concrete case where this makes a difference?  Right now I think there is no way to get a SVG icon into the collection meta data.  Are you working on making that possible?

Does GNOME Software allow cached SVG icons?

File: pkg/apps/watch-appstream.py (hunk 1 above)
Comment (mvollmer) [reply pkg/apps/watch-appstream.py]: Let's not change the cached icon search logic at all, as there is no way in practice to get the collection data generators to create the `/usr/share/app-info/$origin/icons/scalable` directory. I think.

File: pkg/apps/utils.jsx (hunk 2 above)
Comment (mvollmer) [reply pkg/apps/utils.jsx]: I tried removing the `"content-type"` entirely, and PNG icons are still displayed correctly. So I vote to remove it.  Would that work for you as well, @jeepingben?

**Comment by hughsie** [comment]
No, gnome-software can't load svg icons from the cache. The only way we can load svg icons is with an icon type of `remote`.

**Comment by jeepingben** [comment]
I reverted the changes to searching for cached images.  Removing content-type specification prevents svg images from displaying.   I am using my own metadata files and can switch from cached to local icons and it all works as long as the content-type code remains.

File: pkg/apps/utils.jsx (hunk 2 above)
Comment (petervo) [reply pkg/apps/utils.jsx]: How about if we only set content-type if we are loading svgs?

**Comment by mvollmer** [comment]
LGTM now.  Could you squash this to a single commit?  I can also do that when merging.

File: pkg/apps/utils.jsx [hunk 3]
```diff
@@ -43,14 +43,17 @@ export function icon_url(path_or_url) {
...
         payload: "fsread1",
         binary: "raw",
         path: path_or_url,
-        external: {
-            "content-type": "image/png",
-        }
-    }));
+    }
+
+    if (path_or_url.endsWith("svg")) {
```
Comment (martinpitt) [inline pkg/apps/utils.jsx]: For complete correctness this should be  ".svg". If you want to squash, please add the dot as wel.

**Pull request merged**